plugins/kubernetes-common/src/types.ts

```diff
@@ -223,18 +223,46 @@ export interface PodStatusFetchResponse {
 }
 
 /** @public */
-export interface KubernetesFetchError {
+export type KubernetesFetchError =
+  | KubernetesHttpError
+  | KubernetesTimeoutError
+  | KubernetesAddrNotFoundError
+  | KubernetesUnknownError;
+
+/** @public */
+export interface KubernetesHttpError {
   errorType: KubernetesErrorTypes;
   statusCode?: number;
   resourcePath?: string;
 }
 
+/** @public */
+export interface KubernetesTimeoutError {
+  errorType: 'TIMED_OUT';
+  address: string;
+  port?: number;
+}
+
+/** @public */
+export interface KubernetesAddrNotFoundError {
+  errorType: 'ADDR_NOT_FOUND';
+  hostname?: string;
+}
+
+/** @public */
+export interface KubernetesUnknownError {
+  errorType: 'UNKNOWN_ERROR';
+  message: string;
+}
+
 /** @public */
 export type KubernetesErrorTypes =
   | 'BAD_REQUEST'
   | 'UNAUTHORIZED_ERROR'
   | 'NOT_FOUND'
   | 'SYSTEM_ERROR'
+  | 'TIMED_OUT'
+  | 'ADDR_NOT_FOUND'
```

Comment:
Isn't it odd to have these two here? Seems to me that the discriminated union above would have more than one case matching it: e.g. if `errorType` is `'TIMED_OUT'`, then it can be EITHER a `KubernetesHttpError` or a `KubernetesTimeoutError` and you have to additionally sub-filter on other properties. I thought that the new error types were kept separate specifically because you wanted to discriminate on them separated from the old "raw" error type

Reply:
Curious to see what you think of this take: https://github.com/backstage/backstage/pull/15250/commits/0ad476a720f520b267e446f2d64c7cafdd8b5462 -- it's definitely a breaking change in the wire format and hopefully has the data modeling properties you wanted. Sadly I wasn't able to distinguish the different error cases as clearly (i.e. can't distinguish a timeout from an ENOTFOUND) but I think it's worth the tradeoff.